File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        final int MOD = 1234567;

        int a = 0; // F(0)
        int b = 1; // F(1)

        for (int i = 2; i <= n; i++) {
            int next = (a + b) % MOD;
            a = b;
            b = next;
        }

        return b;
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    const MOD = 1234567;

    let a = 0; // F(0)
    let b = 1; // F(1)

    for (let i = 2; i <= n; i++) {
        const next = (a + b) % MOD;
        a = b;
        b = next;
    }

    return b;
}
